modules/acquia_cms_search/tests/src/ExistingSite/AcquiaFacetCreationTest.php

```diff
@@ -0,0 +1,128 @@
+<?php
+
+namespace Drupal\Tests\acquia_cms_search\ExistingSite;
+
+use Drupal\acquia_cms_search\Facade\FacetFacade;
+use Drupal\facets\Entity\Facet;
+use Drupal\taxonomy\Entity\Vocabulary;
+use Drupal\Tests\block\Traits\BlockCreationTrait;
+use Drupal\Tests\taxonomy\Traits\TaxonomyTestTrait;
+use weitzman\DrupalTestTraits\ExistingSiteBase;
+
+/**
+ * Tests Facet Creation and Search Page Block Placement.
+ *
+ * @group acquia_cms
+ * @group acquia_cms_search
+ * @group low_risk
+ * @group pr
+ * @group push
+ */
+class AcquiaFacetCreationTest extends ExistingSiteBase {
+
+  use TaxonomyTestTrait;
+  use BlockCreationTrait;
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp(): void {
+    parent::setUp();
+
+    // Setup the Facets and Create Content to test search page functionality.
+    /** @var \Drupal\taxonomy\VocabularyInterface $vocabulary */
+    $vocabulary = Vocabulary::load('categories');
+    $sub_type = $this->createTerm($vocabulary, ['name' => 'Test Category']);
+    $modules = $this->container->get('module_handler');
+
+    // Create facet & content only if the content type module exists.
+    if ($modules->moduleExists('acquia_cms_person')) {
```

Comment:
There is a better solution than just checking the `acquia_cms_person` module. Doing this, restricts the test to just run only if person content type is installed. 
Instead, we could use a check of the present content types in the system and create content for them all. This will allow the test to run freely with any content type present in the system. Something like this - 

`    $nodeTypes = \Drupal::entityTypeManager()->getStorage('node_type')->loadMultiple();

    if (!empty($nodeTypes)) {
      foreach($nodeTypes as $machine_name => $type) {
        $node = $this->createNode([
          'type' => $machine_name,
          'field_categories' => $sub_type->id(),
          'moderation_state' => 'published',
        ]);
      }
      $facet = Facet::load('test_category_facet');
      if (empty($facet)) {
        // Use the Facade Class to create the facet.
        \Drupal::classResolver(FacetFacade::class)->addFacet('test_category_facet', 'Test Category Facet',
        'category', 'field_categories', 'taxonomy_term_coder', 'search_api:views_page__search__search');
      }
      // Place the facet block in content region to test facets.
      $block = $this->placeBlock('facet_block:test_category_facet', [
        'region' => 'content',
        'id' => 'test_category_facet',
      ]);
      $this->markEntityForCleanup($block);
    }`
    
    Please suggest if this should be done or are we good with just checking the Person content type module.